src/interface/qt/BlockchainSelection.cpp

```diff
@@ -85,6 +85,19 @@ auto BlockchainSelectionQt::enabledCount() const noexcept -> int
     return static_cast<int>(imp_->parent_.EnabledCount());
 }
 
+auto BlockchainSelectionQt::enabledChains() const noexcept -> UnallocatedSet<int>
```

Comment:
I guess, this function is used somewhere else in the code ? I am wondering if is it possible to use as return type UnallocatedSet<Type> ? because Justus sometimes uses some specifice range of integers for types to allow clients use some specific range, and if something will be in the future occupied above integer range we can have not visible at the first glance problem :P 

https://wandbox.org/permlink/hUE03io0oXOZAz5h

If we cannot change it maybe, let's check if type is below numeric_limits<int>::max() in lambda sth like:

static auto const is_above_int_limits = [](const auto value){ return value > std::numeric_limits<int>::max(); }; //that max can be extracted to not call max() each time, the same for lambda if you don't like them :P

static auto const cast_type_f = [](auto const type) { return is_above_int_limits(type) ? some_default_value : static_cast<int>(type);};
UnallocatedSet<int> castedChains;
    std::transform(
        chains.cbegin(),
        chains.cend(),
        std::inserter(castedChains, castedChains.begin()),
        cast_type_f
    );